### DISCUSSION TITLE
How to implement navigation?

### DISCUSSION THREAD

**Discussion by k0m0ri** [eclipse-glsp/glsp#839] [Q&A]
Hello,
I'm using a project-templates node-json-vscode.
I would like to be able to double-click on a Node to jump to the source code (e.g. glsp-client/workspace/example.tasklist) and modify/add parameters (name, etc.) directly.
Therefore, I would like to use [navigation feature](https://www.eclipse.org/glsp/gallery/#navigation).
However, is [navigation of node server feature unchecked](https://github.com/eclipse-glsp/glsp-server-node), meaning it is not available on the node server?

**Comment by tortmayr** [comment]
Hi @k0m0ri ,

>However, is [navigation of node server feature unchecked](https://github.com/eclipse-glsp/glsp-server-node), meaning it is not available on the node server?

In general, an unchecked feature means that it is not yet available for the node server. However, in case of the navigation feature this might not be true. It looks like the required services are at least partly implemented (https://github.com/eclipse-glsp/glsp-server-node/tree/main/packages/server-node/src/features/navigation). We have to test this and will come back to you

**Comment by tortmayr** [comment]
After testing it I can confirm that the full navigation feature set is available for the node server. It seems like somebody forgot to update the feature matrix.

>I would like to be able to double-click on a Node to jump to the source code (e.g. glsp-client/workspace/example.tasklist) and modify/add parameters (name, etc.) directly.

For a reference you can have a look at the [jump to documentation](https://github.com/eclipse-glsp/glsp-server-node/blob/0193b161a5f728fd49176859f04b26f8eb788fa5/examples/workflow-server/src/provider/node-documentation-navigation-target-provider.ts) feature of the workflow example. It jumps to an '.md' file when clicking on the first task node but I guess this can be easily adapted to fit your requirements

**Reply by k0m0ri** [reply]
@tortmayr , thanks for the reply.

Thanks for confirming that it works on the server node.

>For a reference you can have a look at the [jump to documentation](https://github.com/eclipse-glsp/glsp-server->node/blob/0193b161a5f728fd49176859f04b26f8eb788fa5/examples/workflow-server/src/provider/node->documentation-navigation-target-provider.ts) feature of the workflow example. It jumps to an '.md' file when >clicking on the first task node but I guess this can be easily adapted to fit your requirements

If I want to add a jump function to [node-json-vscode](https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/node-json-vscode), do I just implement [the program](https://github.com/eclipse-glsp/glsp-server-node/tree/0193b161a5f728fd49176859f04b26f8eb788fa5/examples/workflow-server/src/provider) up to the fourth from the top in the provider folder?
Also, do I need to add any implementation on the client side?
